File: Algebra_LowestTerm.java
package algebra;

public class Algebra {

    public static void main(String[] args) {

        int numerator;
        int denominator;
        int[] solution;

        System.out.println("Welcome to our demonstration sequels");
        System.out.println("Hope you enjoy (and follow) the lessons.");
        System.out.println("");

        /*
        * Reduce fractions to Lowest Term
         */
        numerator = 16;
        denominator = 640;

        System.out.println("To reduce to lowest term, simplifying:");
        // Print as fraction
        System.out.printf("%45d%n", numerator);
        System.out.printf("%44s%n", "―");
        System.out.printf("%45d%n", denominator);

        // use the LowestTerm class
        LowestTerm red_fract = new LowestTerm(numerator, denominator);
        solution = red_fract.reduceFraction();
        numerator = solution[0];
        denominator = solution[1];

        System.out.println();

        System.out.printf("%46d%n", numerator);
        System.out.printf("%45s%n", "Answer =    ―");
        System.out.printf("%46d%n", denominator);

        System.out.println("\n\n");
    }

}

File: Algebra_LowestTerm.cpp
// Algebra.cpp : Defines the entry point for the console application.
//

#include "stdafx.h"
#include "LowestTerm.h"

#include <iostream>
#include <exception>

using namespace std;

int main()
{
	cout << "\n    Welcome to our demonstration sequels";
	cout << "\n    Hope you enjoy (and follow) the lessons.\n\n";

	unsigned int numerator;
	unsigned int denominator;
	unsigned int whole_number;

	try {
		/*
		* Reduce fractions to Lowest Term
		*/
		numerator = 16;
		denominator = 640;

		cout << "\n    To reduce to lowest term, simplifying:\n";
		// Print as fraction
		printf("%46u\n", numerator);
		printf("%46s\n", "-");
		printf("%46u\n", denominator);

		// use the LowestTerm class
		LowestTerm red_fract(numerator, denominator);
		red_fract.reduceFraction();

		printf("\n%46u\n", red_fract.numerator);
		printf("%46s\n", "Answer =    -");
		printf("%46u\n", red_fract.denominator);

		cout << "\n\n";

	} catch (exception& e) {
		cout << "\n" << e.what() << "\n";
	}

    return 0;
}

